### DISCUSSION TITLE
Berry & Emscripten

### DISCUSSION THREAD

**Discussion by Sebight** [berry-lang/berry#433] [Q&A] [answered]
Hello, I am trying to build our codebase (which Berry has implemented) using Emscripten.

The build process itself is fine, but I am getting runtime errors with invalid pointers.

![image](https://github.com/berry-lang/berry/assets/59232422/c33a7a28-1564-4c8c-bd86-841ab9352618)

It is coming from this function, and I cannot find why and where is this function called. This function is getting an invalid pointer, for some reason. (Ultimately, it ends up in be_string.c, be_newstr(), where it crashes)

The logs indicate the pointer and the string content.
![image](https://github.com/berry-lang/berry/assets/59232422/d3cfe354-bcd8-4ee1-9767-ccb75224e9da)

You can see that the code crashes at the reading of the (invalid pointer) string.

Any idea what could be up? Or maybe prior experience with compiling Berry with Emscripten? I suppose @skiars had to undergo this for the website Berry execution? 

Thanks!

**Comment by s-hadinger** [comment]
I have built it several times without any issue. Did you use this repository ?

https://github.com/berry-lang/berry_web

**Reply by Sebight** [reply]
Nope. I am compiling the whole C++ codebase with "normal" Berry. What is the point of berry_web? Differences in `port/` files? Or rather the `platform/` files.

**Reply by s-hadinger** [reply]
Yes. I always start from berry_web when compiling for emscripten. I'm not sure of all differences, but it's always easier to start from a working example. @skiars may provide more information

**Reply by Sebight** [reply]
I see that the port/ files are different. Depending on the platform I'm currently compiling for, I might be able to hot-swap those. I'll try that out.

**Reply by s-hadinger** [reply]
Happy to get feedback from you on how this evolves

**Reply by Sebight** [reply, marked as answer]
This fixed the entirety of my problem. 😄 

![image](https://github.com/berry-lang/berry/assets/59232422/2fde4283-70a2-4ecd-bf09-40a36098a7c9)

Probably not a good practice, but I will test out if it affects anything in my ESP32 environment.
